addons/pos_loyalty/models/pos_config.py

```diff
@@ -63,6 +63,12 @@ def _check_before_creating_new_session(self):
                 reward = gc_program.reward_ids
                 if reward.reward_type != 'discount' or reward.discount_mode != 'per_point' or reward.discount != 1:
                     raise UserError(_('Invalid gift card program reward. Use 1 currency per point discount.'))
+                if self.gift_card_settings == "create_set":
+                    if not gc_program.mail_template_id:
+                        raise UserError(_('There is no email template on the gift card program.'))
```

Comment:
You can specify both error message by adding "and your pos is set to print them" (or send them by email)